controllers/numaresourcesoperator_controller.go

```diff
@@ -286,13 +292,13 @@ func (r *NUMAResourcesOperatorReconciler) reconcileResource(ctx context.Context,
 	return ctrl.Result{}, status.ConditionAvailable, nil
 }
 
-func (r *NUMAResourcesOperatorReconciler) syncDaemonSetsStatuses(ctx context.Context, rd client.Reader, daemonSetsInfo map[string]nropv1.NamespacedName) ([]nropv1.NamespacedName, bool, error) {
+func (r *NUMAResourcesOperatorReconciler) syncDaemonSetsStatuses(ctx context.Context, rd client.Reader, daemonSetsInfo []poolDaemonSet) ([]nropv1.NamespacedName, bool, error) {
 	dssWithReadyStatus := []nropv1.NamespacedName{}
-	for _, nname := range daemonSetsInfo {
+	for _, dsInfo := range daemonSetsInfo {
 		ds := appsv1.DaemonSet{}
 		dsKey := client.ObjectKey{
```

Comment:
this will then become: `dsKey := nname.AsKey()` (maybe inlined, seems to be used only once)